default/scripting/techs/defense/Defense.focs.txt

```diff
@@ -26,7 +26,7 @@ Tech
             effects = [
                 SetShield value = min(Value + [[PLANET_SHIELD_FACTOR]], Value(Target.MaxShield))
                 SetDefense value = min(Value + [[PLANET_DEFENSE_FACTOR]], Value(Target.MaxDefense))
-                SetTroops value = min(Value + 1, Value(Target.MaxTroops))
+                SetTroops value = min(max(Value + 1, 6), Value(Target.MaxTroops))
```

Comment:
this always regenerates to a minimum of six troops. certainly not intended. setting to six should be gated by colony age / current turn.

Reply:
It is intended, it is indeed the whole point of this PR: to make "gifting through fake conquest" more expensive than a single small hull troop ship.
Without this change, empires with the first troop tech start their colonies at 2+. This is making it 6+, without really increasing target meter mid game, so not so different. This also makes more expensive conquering genuinely undefended colonies, but that I also see that as an improvement: starting a game next to an inexperienced player will no longer be such an advantage against other players. Always keeping in mind that the increase of cost (from the smallest troop ship to 2-4x that cost depending on several factors) is quite modest.

Reply:
This ensures 6 troops even if you were reduced to less by invasion

Reply:
The effect only applies if planet wasn't attacked last turn: `LocalCandidate.LastTurnAttackedByShip   < CurrentTurn`

I'll try to gate the six part of the effect through something like LastTurnColonized (since there isn't anything like LastTurnGroundCombat) and report back.

Reply:
LastTurnColonized seems to work well. The instantaneous min. 6 troops regen. applies after colonization and never again. I need to test it for combat situations before resolving this conversation.

Reply:
This doesn't work:
```
        EffectsGroup    // set troops to 6 if just colonized
            scope = And [
                Planet
                OwnedBy empire = Source.Owner
                (LocalCandidate.LastTurnColonized = CurrentTurn)
            ]
            priority = [[AFTER_ALL_TARGET_MAX_METERS_PRIORITY]]
            effects = SetTroops value = min((NamedRealLookup name = "IMPERIAL_GARRISON_MAXTROOPS_EFFECT"), Value(Target.MaxTroops))
```
Turn X: A orders invasion of planet owned by B.
Turn X+1: B invades planet. A fleet has arrived at planet and combat happened, but troops of planet are 6.2.

Reply:
If I read correctly, the troops are 6.2 after invasion?

In that case I would guess that LastTurnColonized is broken? Have look at the Planet/"Last Turn Colonized" column in the  objects window.

You could also add e.g. a SetMaxResearch effect to that effectsgroup for debugging purposes (so you can see the effect actually fired).

Reply:
Looking at the code I would say the last turn colonized is fine. Maybe LocalCandidate is broken (I already opened an issue for some weirdness where a fix should not have worked). Can you see if there was any planet in that universe colonized in that turn (objects window)?

Reply:
The SetTroops quoted above only triggers for new colonies. Checked out using SitReps. Invasion doesn't trigger the effect. So the extra troops regeneration on first turn after invasion is coming from somewhere else. Working on that.

Reply:
[Discussion thread](https://www.freeorion.org/forum/viewtopic.php?p=103814#p103814).

Reply:
Ending this conversation:
- The newly added minimum of 6 troops only triggers the turn of colonization, as intended.
- The weird behaviour for just-conquered planets is solved by not allowing Self-defense's Shield, Defense and Troops regeneration the turn they are conquered (same it is not allowed when attacked by ships).
- This solution is preferred to altering backend to ensure always construction one before combats.
- Playtested and working as intended.